pagespeed/system/system_test.sh

```diff
@@ -2249,3 +2249,10 @@ if [ "$CACHE_FLUSH_TEST" = "on" ]; then
   # check [ $bytes -gt 100000 ]
   check [ $bytes -lt 100000 ]
 fi
+
+start_test AddResourceHeaders works for pagespeed resources.
+URL="$TEST_ROOT/compressed/hello_js.custom_ext.pagespeed.ce.HdziXmtLIV.txt"
+HTML_HEADERS=$($WGET_DUMP $URL)
+# TODO(oschaaf): a helper to perform this type of matching.
+MATCHES=$(echo "$HTML_HEADERS" | grep -c "^X-Foo: Bar") || true
```

Comment:
how about:
  check_from "$HTML_HEADERS" grep -q "^X-Foo: Bar"
See also extract_headers, which will give you just the headers, but it requires a file.

The advantage of check_from is that if it fails, it prints what the contents of the variable in the failure message.  This is important for flaky tests, where if you simply try it again it will not necessarily fail, or fail the same way.